lib/grpc_reflection/service/builder.ex

```diff
@@ -198,4 +279,102 @@ defmodule GrpcReflection.Service.Builder do
   end
 
   defp upcase_first(<<first::utf8, rest::binary>>), do: String.upcase(<<first::utf8>>) <> rest
+
+  # generate a field descriptor from a field props struct
+  defp convert_to_field_descriptor(
+         extendee,
+         %Protobuf.Extension.Props.Extension{field_props: field_props}
+       ) do
+    {type, type_name} = type_from_field_props(field_props)
+
+    %Google.Protobuf.FieldDescriptorProto{
+      name: field_props.name,
+      number: field_props.fnum,
+      label: label_from_field_props(field_props),
+      type: type,
+      type_name: type_name,
+      extendee: extendee
+    }
+  end
+
+  # Google.Protobuf.FieldDescriptorProto.Label
+  defp label_from_field_props(%Protobuf.FieldProps{optional?: true}), do: 1
+  defp label_from_field_props(%Protobuf.FieldProps{repeated?: true}), do: 3
+  defp label_from_field_props(%Protobuf.FieldProps{required?: true}), do: 2
+
+  # Google.Protobuf.FieldDescriptorProto.Type
+  defp type_from_field_props(%Protobuf.FieldProps{type: type}) do
```

Comment:
running credo locally:

```
 [F] → Function is too complex (cyclomatic complexity is 23, max is 9).
┃       lib/grpc_reflection/service/builder.ex:306:8 #(GrpcReflection.Service.Builder.type_from_field_props)
```
This function looks like it would behave well as a pattern match

```
  # Google.Protobuf.FieldDescriptorProto.Type
  defp type_from_field_props(%Protobuf.FieldProps{type: :double}), do: {1, nil}
  defp type_from_field_props(%Protobuf.FieldProps{type: :float}), do: {2, nil}
  defp type_from_field_props(%Protobuf.FieldProps{type: :int64}), do: {3, nil}
```

Reply:
I refined them by match pattern.